buffer/src/main/java/io/netty/buffer/UnpooledByteBufAllocator.java

```diff
@@ -46,14 +62,23 @@ protected ByteBuf newHeapBuffer(int initialCapacity, int maxCapacity) {
 
     @Override
     protected ByteBuf newDirectBuffer(int initialCapacity, int maxCapacity) {
-        ByteBuf buf;
-        if (PlatformDependent.hasUnsafe()) {
-            buf = new UnpooledUnsafeDirectByteBuf(this, initialCapacity, maxCapacity);
-        } else {
-            buf = new UnpooledDirectByteBuf(this, initialCapacity, maxCapacity);
-        }
-
-        return toLeakAwareBuffer(buf);
+        ByteBuf buf = PlatformDependent.hasUnsafe() ?
+                new UnpooledUnsafeDirectByteBuf(this, initialCapacity, maxCapacity) :
+                new UnpooledDirectByteBuf(this, initialCapacity, maxCapacity);
+
+        return disableLeakDetector ? buf : toLeakAwareBuffer(buf);
```

Comment:
i see ... so accepting the fact that GC may never run to clean up direct buffers is desired in this case?